gym/lib/gym/runner.rb

```diff
@@ -11,26 +11,36 @@ def run
         clear_old_files
         build_app
       end
-      verify_archive
-
+      verify_archive if Gym.project.produces_archive?
       FileUtils.mkdir_p(File.expand_path(Gym.config[:output_directory]))
 
       if Gym.project.ios? || Gym.project.tvos?
         fix_generic_archive # See https://github.com/fastlane/fastlane/pull/4325
-        package_app
+        package_app if Gym.project.produces_archive?
         fix_package
         compress_and_move_dsym
-        path = move_ipa
+        path = move_ipa if Gym.project.produces_archive?
         move_manifest
         move_app_thinning
         move_app_thinning_size_report
         move_apps_folder
-
-        path
       elsif Gym.project.mac?
         compress_and_move_dsym
-        copy_mac_app
+        copy_mac_app if Gym.project.mac_app?
```

Comment:
Do we want to continue doing this work if it is a mac app? Shouldn't we just end this as we did before? I think I'm confused about what this does after it copies the Mac app.